MJK playground updates to SDF 1.1

mjkoster opened the pull request:
update the Bluetooth mesh and Zigbee examples to use SDF 1.1 patterns

mjkoster committed:
update Bluetooth Mesh models to 1.1

level and onoff

mjkoster committed:
Update sdfobject-genericdefaulttransitiontime.sdf.json

sdf 1.1 updates

mjkoster committed:
update zigbee examples

sdf 1.1

cabo commented on sdfObject/sdfobject-onoff.sdf.json:
```diff
@@ -111,16 +107,18 @@
           "minimum": 0, 
           "maximum": 65535, 
-          "multipleOf": 1
+          "multipleOf": 1,
```
We do have integer, wouldn't that be a more natural way to say this than the combination of number and multipleof=1?

mjkoster replied:
I made all of these cases "integer" to be consistent

akeranen commented on sdfObject/sdfobject-genericlevel.sdf.json:
```diff
@@ -124,19 +118,22 @@
-          "multipleOf": 1
+          "multipleOf": 1,
+          "scaleMinimum": 0,
+          "scaleMaximum": 1.275
```
What is the difference between scaleMax and max? Seems underspecified in the SDF spec

mjkoster replied:
removed the scale numbers since they are not in 1.1

mjkoster committed:
Update sdfobject-level.sdf.json

missed a 1.1 conversion in the level object

cabo commented on sdfObject/sdfobject-genericdefaulttransitiontime.sdf.json:
```diff
@@ -1,7 +1,7 @@
   "info": {
     "title": "Example Bluetooth mesh Generic Default Transition Time Model", 
-    "version": "20200710", 
+    "version": "20210-03-05", 
```
While version strings are not well-defined in SDF 1.1, this looks like a typo.

mjkoster replied:
fixed

cabo commented on sdfObject/sdfobject-genericdefaulttransitiontime.sdf.json:
```diff
@@ -22,27 +22,29 @@
       "sdfAction": {
```
These look a lot like properties to me.

mjkoster replied:
I made a standalone sdfData definition to make these reusable as properties of an object type sdfProperty. We can compare to the generic level control, where the data structure has 2 more fields added. After we merge?

cabo replied:
This is certainly not incorrect SDF, but it strikes me as an abuse of sdfAction to just serve as getter/setter pairs -- this is what properties were meant for.

cabo commented on sdfObject/sdfobject-genericlevel.sdf.json:
```diff
@@ -124,19 +118,22 @@
           "maximum": 255,
-          "multipleOf": 1
+          "multipleOf": 1,
+          "scaleMinimum": 0,
```
No longer part of SDF 1.1

mjkoster replied:
fixed all of these

cabo replied:
I still find a couple in sdfObject/sdfobject-genericonoff.sdf.json...

mjkoster replied:
Removed

mjkoster committed:
Addressed comments on PR

updated to remove scaleMinimum, etc.
added a standalone sdfData definition for the Bluetooth Mesh example

cabo commented on sdfObject/sdfdata-genericdefaulttransitiontime.sdf.json:
```diff
@@ -0,0 +1,57 @@
+      "description": "data elements packed into an array to represent the Bluetooth Mesh state",
+      "type": "array",
+      "uniqueItems": true,
+      "items": [
```
We currently do not allow inhomogeneous arrays: This is what `"type": "object"` is meant for in 1.1

mjkoster replied:
Added type: integer to the sdfChoice construct so both array items are now integer

cabo commented on sdfObject/sdfobject-level.sdf.json:
```diff
@@ -85,138 +85,135 @@
-            "SetToPreviousValue"
-          ]
+          "label": "StartUpCurrentLevel",
+          "type": "integer", 
```
sdfChoice currently does not allow to "factor out" data shape qualities such as type etc.
Should it?

mjkoster replied:
I believe we added this to the current schema

JKRhb commented:
If I see it correctly the biggest obstacle for a merge of this PR is the inhomogeneous array in the newly added model, right? Would it be possible to already merge the updates to the existing models and move the new one to a separate PR?

mjkoster committed:
Merge branch 'master' into mjk-pg-1-1-update

mjkoster committed:
clean up v1.0 issues

mjkoster committed:
Update sdfobject-level.sdf.json

validation schema seems not to accept type on sdfChoice

mjkoster committed:
Update sdfobject-level.sdf.json

mjkoster committed:
Update sdfdata-genericdefaulttransitiontime.sdf.json

mjkoster committed:
Update sdfdata-genericdefaulttransitiontime.sdf.json

mjkoster committed:
Update sdfdata-genericdefaulttransitiontime.sdf.json

work around for tuple pattern
